alu_u32/src/shift/stark.rs

```diff
@@ -0,0 +1,62 @@
+use super::columns::Shift32Cols;
+use super::Shift32Chip;
+use core::borrow::Borrow;
+
+use p3_air::{Air, AirBuilder};
+use p3_field::AbstractField;
+use p3_matrix::MatrixRowSlices;
+
+impl<F, AB> Air<AB> for Shift32Chip
+where
+    F: AbstractField,
+    AB: AirBuilder<F = F>,
+{
+    fn eval(&self, builder: &mut AB) {
+        let main = builder.main();
+        let local: &Shift32Cols<AB::Var> = main.row_slice(0).borrow();
```

Comment:
why outputs from Shift32Cols never constrained here?

Reply:
I think `output` should be implicitly constrained by the bus messages; i.e. this chip essentially calls out to either the mul or div chip to get the result

Reply:
I tried a small tests and constraints fail https://github.com/vivekvpandya/valida/commit/21a90fdda80d4fb969f82dd3f9a97642fb5ef366